packages/babel-core/src/transformation/block-hoist-plugin.js

```diff
@@ -41,23 +39,29 @@ const blockHoistPlugin = {
   visitor: {
     Block: {
       exit({ node }) {
-        let hasChange = false;
-        for (let i = 0; i < node.body.length; i++) {
-          const bodyNode = node.body[i];
-          if (bodyNode?._blockHoist != null) {
-            hasChange = true;
-            break;
+        if (!node.body.some(node => node?._blockHoist != null)) return;
+        // TODO: Babel 9 (?) - remove stabilityMap once Node >= 12 is guaranteed
+        // Array sorting is not stable in earlier Node releases
+        // See: https://v8.dev/blog/array-sort for details
+        const nodeVersion =
+          typeof process !== "undefined"
+            ? process.versions.node.split(".")[0]
+            : undefined;
+        const stabilityMap =
+          !nodeVersion || nodeVersion < 12
```

Comment:
```suggestion
            : 0;
        const stabilityMap =
          nodeVersion < 12
```

Feel free to ignore this suggestion, just wanted to point out that the conditional can be simplified if you make browsers produce `nodeVersion === 0`.

Reply:
Thanks @coreyfarrell - I think I'd prefer to keep the meaning/semantics of `nodeVersion` closer to the ground truth (i.e. it'll genuinely have an undefined value in non-Node environments), even though I'd admit that it's (hopefully) unlikely that anyone would be line-by-line debugging the value.  I'm sure I've probably used tricks like this to simplify my own code elsewhere too, so it's a bit arbitrary of me not to accept.  But I think I'd prefer the more verbose condition in this case.